lamindb/models/query_set.py

```diff
@@ -668,9 +692,13 @@ def extract_single_element(s):
         result = reorder_subset_columns_in_df(result, feature_names)
 
     if cols_from_include:
-        result = process_cols_from_include(df, result, cols_from_include)
-
-    return result.drop_duplicates(subset=["id"])
+        result = process_cols_from_include(df_encoded, result, cols_from_include)
+    # here we revert the field encoding
+    # but in cases when the result already has a column with the field name (like a feature also called 'id'),
```

Comment:
This unpredictable magic is a problem...

I'd say that if `features=True` we should always encode. But if `features=False` we should never encode.

Something like this. 🤔